Formula/cadence-workflow.rb

```diff
@@ -0,0 +1,28 @@
+class CadenceWorkflow < Formula
+  desc "Distributed, scalable, durable, and highly available orchestration engine"
+  homepage "https://cadenceworkflow.io/"
+  url "https://github.com/uber/cadence/archive/refs/tags/v0.18.2.tar.gz"
+  sha256 "1cdffbee29d9455a0ff4e61114f93b1f5fc210366b707b7dd496a90325a21dce"
+  license "MIT"
+  head "https://github.com/uber/cadence.git"
+
+  depends_on "go" => :build
+
+  conflicts_with "cadence", because: "both install an `cadence` executable"
+
+  def install
+    system "make", "cadence", "cadence-server", "cadence-canary", "cadence-sql-tool", "cadence-cassandra-tool"
+    bin.install "cadence"
+    bin.install "cadence-server"
+    bin.install "cadence-canary"
+    bin.install "cadence-sql-tool"
+    bin.install "cadence-cassandra-tool"
+
+    (etc/"cadence").install "config", "schema"
+  end
+
+  test do
+    assert_match "cadence version 0.18.3",
+      shell_output("#{bin}/cadence --version")
```

Comment:
We still need a test more substantial than this. It has to exercise some functionality.

Please see the [docs](https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula) for more detail on this. Quoting from there:
>* If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
>A good example is [`tinyxml2`](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/tinyxml2.rb), which writes a small C++ source file into the test directory, compiles and links it against the tinyxml2 library and finally checks that the resulting program runs successfully.
>* If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
>* If the software cannot function without credentials or requires a virtual machine, docker instance, etc. to run, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected. This is prefered over mocking a dependency.
>* Homebrew comes with a number of [standard test fixtures](https://github.com/Homebrew/brew/tree/master/Library/Homebrew/test/support/fixtures), including numerous sample images, sounds, and documents in various formats. You can get the file path to a test fixture with `test_fixtures("test.svg")`.

Reply:
cadence-server will need to run with databases(MySQL/Postgers or Cassandra), and cadence/cadence-canary/etc(the CLIs) will need to run with server. 

So I think I will just verify the failure case. Does it work for you? 

So without DB, the start would fail like this:
```
qlong@~/cadence:
(master)$./cadence-server start
2021/04/05 18:16:46 Loading config; env=development,zone=,configDir=./config
2021/04/05 18:16:46 Loading configFiles=[./config/base.yaml ./config/development.yaml]
2021/04/05 18:16:46 gocql: unable to dial control conn 127.0.0.1: dial tcp 127.0.0.1:9042: connect: connection refused
2021/04/05 18:16:46 cassandra schema version compatibility check failed: unable to create CQL Client: gocql: unable to create session: control: unable to connect to initial hosts: dial tcp 127.0.0.1:9042: connect: connection refused
```

Reply:
That's fine, thanks.

Reply:
Added. Could you take another look? Thanks